suite-common/wallet-core/src/fiat-rates/fiatRatesThunks.ts

```diff
@@ -0,0 +1,364 @@
+import { differenceInMilliseconds, getUnixTime, subWeeks } from 'date-fns';
+
+import {
+    fetchCurrentFiatRates,
+    fetchCurrentTokenFiatRates,
+    fetchLastWeekRates,
+    getFiatRatesForTimestamps,
+} from '@suite-common/fiat-services';
+import TrezorConnect, { AccountTransaction } from '@trezor/connect';
+import { createThunk } from '@suite-common/redux-utils';
+import { Network, networksCompatibility as NETWORKS } from '@suite-common/wallet-config';
+import { Account, CoinFiatRates, TickerId } from '@suite-common/wallet-types';
+import { FIAT } from '@suite-common/suite-config';
+import { getAccountTransactions, isTestnet } from '@suite-common/wallet-utils';
+import { getBlockbookSafeTime } from '@suite-common/suite-utils';
+
+import { fiatRatesActions, actionPrefix } from './fiatRatesActions';
+import { selectCoins } from './fiatRatesReducer';
+import { selectAccounts } from '../accounts/accountsReducer';
+
+let staleRatesTimeout: ReturnType<typeof setInterval>;
+let lastWeekTimeout: ReturnType<typeof setInterval>;
+
+// how often should suite check for outdated rates;
+const INTERVAL = 1000 * 60 * 2; // 2 mins
+const INTERVAL_LAST_WEEK = 1000 * 60 * 60 * 1; // 1 hour
+// which rates should be considered outdated and updated;
+const MAX_AGE = 1000 * 60 * 10; // 10 mins
+const MAX_AGE_LAST_WEEK = 1000 * 60 * 60 * 1; // 1 hour
+
+export const removeFiatRatesForDisabledNetworksThunk = createThunk(
+    `${actionPrefix}/removeRatesForDisabledNetworks`,
+    (_, { dispatch, extra, getState }) => {
+        const {
+            selectors: { selectEnabledNetworks },
+        } = extra;
+        const enabledNetworks = selectEnabledNetworks(getState());
+        const fiat = selectCoins(getState());
+        fiat.forEach(f => {
+            const rateNetwork = (f.mainNetworkSymbol ?? f.symbol) as Network['symbol'];
+            if (!enabledNetworks.includes(rateNetwork)) {
+                dispatch(fiatRatesActions.removeFiatRate(f));
+            }
+        });
+    },
+);
+
+/**
+ *  Returns an array with coin tickers that need fullfil several conditions.
+ *  Array of coin tickers is combined from 2 sources - config file and fiat reducer itself
+ *  Conditions:
+ *  1. network for a given ticker needs to be enabled
+ *  2a. no rates available yet (first fetch)
+ *  OR
+ *  2b. duration since the last check is greater than passed `interval`
+ *  Timestamp is extracted via `timestampFunc`.
+ *
+ * @param {((ticker: CoinFiatRates) => number | undefined)} timestampFunc
```

Comment:
Please remove these JSDocs params comments everywhere in this file or update them to be valid (now there is only one object param, with these inside).